marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -60,19 +66,21 @@ function addPropertiesToSearchResponse(entityName, searchResponse) {
       }
     }
   });
-
   // Make it easy for the client to know which property names were used, and which ones exist
+  // searchResponse.selectedPropertyDefinitions = selectedPropertyMetadata;
   searchResponse.selectedPropertyDefinitions = selectedPropertyMetadata;
   searchResponse.entityPropertyDefinitions = propertyMetadata;
 }
 
-function buildPropertyMetadata(parentPropertyName, entityModel, entityName) {
+// This function builds the logical entityType property metadata for all entityType properties from an entityModel.
+function buildAllMetadata(parentPropertyName, entityModel, entityName) {
   const entityType = entityModel.definitions[entityName];
   if (!entityType) {
     ds.throwServerError("Could not build property metadata; could not find entity type with name: " + entityName);
   }
 
-  const allPropertyMetadata = [];
+  const allPropertiesMetadata = [];
+  let granularPropertyMetadata = {};
```

Comment:
What does "granular" mean in this context? Is it really "selectedPropertyMetadata"?

Reply:
`{
              "propertyPath": "billing.zip.plusFour",
              "propertyLabel": "plusFour",
              "datatype": "string",
              "multiple": false
            }`
The parent structure from where it is referred to is ignored. But its child properties do exist in the object if the object has them